language-server/modules/langserver-core/src/test/resources/completion/expression_context/config/mapping_expr_ctx_config54.json

```diff
@@ -4,52 +4,5 @@
     "character": 28
   },
   "source": "expression_context/source/mapping_expr_ctx_source54.bal",
-  "items": [
-    {
-      "label": "a",
-      "kind": "Variable",
-      "detail": "int",
-      "sortText": "AB",
-      "insertText": "a",
-      "insertTextFormat": "Snippet"
-    },
-    {
-      "label": "CONST1",
-      "kind": "Variable",
-      "detail": "\"Test\"",
-      "documentation": {
-        "right": {
-          "kind": "markdown",
-          "value": ""
-        }
-      },
-      "sortText": "AB",
-      "insertText": "CONST1",
-      "insertTextFormat": "Snippet"
-    },
-    {
-      "label": "param2",
-      "kind": "Variable",
-      "detail": "int",
-      "sortText": "AB",
-      "insertText": "param2",
-      "insertTextFormat": "Snippet"
-    },
-    {
-      "label": "param1",
-      "kind": "Variable",
-      "detail": "int",
-      "sortText": "AB",
-      "insertText": "param1",
-      "insertTextFormat": "Snippet"
-    },
-    {
-      "label": "b",
-      "kind": "Variable",
-      "detail": "map<any>",
-      "sortText": "AB",
-      "insertText": "b",
-      "insertTextFormat": "Snippet"
-    }
-  ]
+  "items": []
```

Comment:
Shall we check this test case? We should provide completions in this context

Reply:
In this case, the source code is:
```ballerina
import ballerina/module1;

public const CONST1 = "Test";

public function testFunction(int param1, int param2) {
    int a = 5;
    map<any> b = {readonly a<cursor>};
}
```
At `<cursor>`, what user should be writing is the name of the key. And I'm not sure if `readonly` has a meaning inside a map. But we can write:
```ballerina
    map<any> b = {
        readonly key1: ""
    };
```

Therefore, AFAIU, not suggesting anything here is correct. WDYT?

Reply:
IINM, the following is also allowed by the compiler according to the spec.
https://ballerina.io/spec/lang/master/#mapping-constructor-expr. Check the rule `[readonly] variableName`
```
public const CONST1 = "Test";

public function testFunction(int param1, int param2) {
    int a = 5;
    map<any> b = {readonly a};
}
```

Reply:
Isn't it a `mapping-binding-pattern`? https://ballerina.io/spec/lang/master/#mapping-binding-pattern

Reply:
IINM, the mapping binding pattern come at the LHS of a variable def. That's also a possible use-case but we should capture it separately. Please correct me if I have been mistaken.

Reply:
We have to support `[readonly] variableName` inside mapping constructor. So, this test case need to fixed. I will look into that.

Reply:
Yeah agree, Context Type Resolver finds the context incorrectly for the scenario as `RecordType` where it has to be `AnyType`

Reply:
@gayalkuruppu that's expected. Because the RHS is a record in this scenario.

@nadeeshaan I had a look at this. Issue is, when providing variable completions, we look at the missing field names in the mapping constructor and suggest variables which match the names of missing fields. So, what we are getting here is correct. (Field `a` is already added in `map<any> b = {readonly a};`. RHS is a `record { int a; ...}`).

We have to suggest variable names regardless of them matching field names IMO. And we can address that in a new issue. WDYT?